pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -1189,6 +1193,124 @@ func (c *Cluster) saveMonConfig() error {
 }
 
 func (c *Cluster) persistExpectedMonDaemons() error {
+	if err := c.persistExpectedMonDaemonsInConfigMap(); err != nil {
+		return errors.Wrap(err, "failed to persist expected mons in ConfigMap")
+	}
+
+	if err := c.persistExpectedMonDaemonsAsEndpointSlice(); err != nil {
+		return errors.Wrap(err, "failed to persist expected mons in EndpointSlice")
+	}
+	return nil
+}
+
+func (c *Cluster) persistExpectedMonDaemonsAsEndpointSlice() error {
+	monitors := c.ClusterInfo.AllMonitors()
+	var monAddressesIPv4, monAddressesIPv6 []string
+
+	if len(monitors) == 0 {
+		logger.Debug("no mon addresses found, skipping endpointslice resource creation")
+		return nil
+	}
+
+	for _, mon := range monitors {
+		ipPort := mon.Endpoint
+
+		var ip string
+		if strings.HasPrefix(ipPort, "[") {
+			ip = ipPort[1:strings.Index(ipPort, "]")]
+		} else {
+			ip = strings.Split(ipPort, ":")[0]
+		}
+
+		parsedIP := net.ParseIP(ip)
+		if parsedIP == nil {
+			logger.Warningf("invalid IP address found: %s", ip)
+			continue
+		}
+
+		if parsedIP.To4() != nil {
+			monAddressesIPv4 = append(monAddressesIPv4, ip)
+		} else {
+			monAddressesIPv6 = append(monAddressesIPv6, ip)
+		}
+	}
+
+	createIPv4 := c.spec.Network.IPFamily == cephv1.IPv4 || c.spec.Network.DualStack
+	logger.Debugf("check if should create IPv4 endpointslice: %v (IPFamily=%s, DualStack=%v)",
+		createIPv4, c.spec.Network.IPFamily, c.spec.Network.DualStack,
+	)
+	createIPv6 := c.spec.Network.IPFamily == cephv1.IPv6 || c.spec.Network.DualStack
+	logger.Debugf("check if should create IPv6 endpointslice: %v (IPFamily=%s, DualStack=%v)",
+		createIPv6, c.spec.Network.IPFamily, c.spec.Network.DualStack,
+	)
+
+	// Default to IPv4 if no network settings are provided.
+	// This ensures backward compatibility with older CephCluster configurations,
+	// and prevents unexpected failures where no EndpointSlice would be created.
+	if c.spec.Network.IPFamily == "" {
+		logger.Debug("no IP family explicitly set, forcing IPv4 mode")
+		createIPv4 = true
+	}
+
+	if createIPv4 && len(monAddressesIPv4) > 0 {
+		if err := c.createOrUpdateEndpointSlice(monAddressesIPv4, "IPv4"); err != nil {
+			return err
+		}
+	}
+
+	if createIPv6 && len(monAddressesIPv6) > 0 {
+		if err := c.createOrUpdateEndpointSlice(monAddressesIPv6, "IPv6"); err != nil {
+			return err
+		}
+	}
```

Comment:
Currently, EndpointSlice updates are missed in the following cases:

- If one (or more) IPv4 mon endpoints existed in the past, but now only IPv6 mon endpoints exist, the former IPv4 endpoints are not removed from the IPv4 EndpointSlice. Solution: simply drop the check `len(monAddressesIPv4) > 0` and let `createOrUpdateEndpointSlice` write an empty list.
- If createIPv4 was true in the past and now is false, the IPv4 EndpointSlice is leaked.
- The same two cases symmetrically with IPv6.